staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -209,6 +209,12 @@ type BIOS struct {
 // ---
 // +k8s:openapi-gen=true
 type EFI struct {
+	// If set, SecureBoot will be enabled and the OVMF roms will be swapped for
+	// SecureBoot-enabled ones.
+	// Requires SMM to be enabled.
```

Comment:
Since this requires SMM, which one can explicitly enable or disable, could we do the following:

In the defaulter:
 * If secureBoot is set to true and SMM is unset (neither explicit true or false), enable SMM too.

In the validator:
 * If secureBoot is set to true and SSM is explicitlly disabled, reject the VM/VMI/VMIRS?

Reply:
The validator part is definitely missing, I'll add it.
However I don't like the idea of defaulting SMM to true without the user explicitly enabling it. SMM is quite a ... dangerous feature.
Thanks for the review!

Reply:
> SMM is quite a ... dangerous feature.

Could you elaborate on that? I can understand that applications inside the guest may attack SMM RAM and succeed, but I don't understand the difference to not having protected RAM at all. Would be happy to learn about the dangers.

Reply:
I believe that if the application in question manages to write malicious code into SMM RAM and then trigger an SMI, there could now be an invisible rootkit controlling the VM.
Multiple security features are in place to prevent all that, but it still seems to me that enabling SMM slightly increases the attack surface of the VM.
Most likely not a problem here, but it's still worth making sure the user knows it's there IMO.

Reply:
> I believe that if the application in question manages to write malicious code into SMM RAM and then trigger an SMI, there could now be an invisible rootkit controlling the VM.
> Multiple security features are in place to prevent all that, but it still seems to me that enabling SMM slightly increases the attack surface of the VM.
> Most likely not a problem here, but it's still worth making sure the user knows it's there IMO.

Thanks, so let's continue without ssm auto-enable for now. I don't know how many signed firmwares requires SMM too. I think that if most do, then it would still make sense to enable it automatically, but let's wait for user-feedback. If no one complains it is definitely good enough.